### PR TITLE
HYB-877: Add preview to scaffold

### PR DESCRIPTION
**Pull request by kerrmarin**
Add dynamic preview to scaffold

JIRA: https://mobify.atlassian.net/browse/HYB-877
Linked PRs: https://github.com/mobify/astro/pull/622

## Changes
- Use previewController to enable or disable preview on app startup

## How to test-drive this PR
- Point astro to HYB-877-dynamic-preview
- Enable preview in BaseConfig
- Run app and shake shake shake 🎶 🎉 

## TODOS:
- [x] Change works in both Android and iOS.
- [x] CHANGELOG.md has been updated.
- [x] Run the generator to make sure it still functions correctly.
- [ ] +1 from an engineer on the Astro team.
- [x] Both tab layout and drawer layout are fully functional in ios. (Set `useTabLayout` in `baseConfig`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "author": "Mobify",
   "dependencies": {
-    "astro-sdk": "mobify/astro.git#develop",
+    "astro-sdk": "mobify/astro.git#eded8186a3a9e86462ad0b509ef0b4293254eb68",
```
Comment (kerrmarin) [inline package.json]: Reminder to move this back to develop before merging

**Comment by coffee-cup** [comment]
I am unable to get this to build. When I open in XCode 7 I get an error saying the document requires XCode 8 or later and when opening in XCode 8 it asks me to convert to the latest Swift syntax. I could be doing something wrong and will talk with you on Monday :)

**Comment by kerrmarin** [comment]
👍 I think it might be that you just need to run an `npm install`, but yea, let's talk Monday :)

File: app/app.js [hunk 2]
```diff
@@ -158,10 +160,21 @@ window.run = function() {
...
-        BaseConfig.previewEnabled ? runAppPreview() : runApp();
+        PreviewController.init().then(function(previewController) {
+            Application.on('previewToggled', function() {
+                previewController.presentPreviewAlert();
+            });
 
+            previewController.isPreviewEnabled().then(function(enabled) {
+                // Configure the previewEnabled flag located in baseConfig.js
+                // to enable/disable app preview
+                if (enabled === true && BaseConfig.previewEnabled) {
```
Comment (coffee-cup) [inline app/app.js]: Can this just be `if (enabled && BaseConfig.previewEnabled)`?

**Comment by jeremywiebe** [comment]
Nice.  Pumped this is now in the scaffold!

**Pull request merged**